presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -266,6 +263,25 @@ default void finishDelete(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Notifies when select query starts
+     *
+     * @return serializable {@link ConnectorTableHandle}. This handle will be brought back in {@link #finishSelect(ConnectorSession, ConnectorTableHandle)} if null returned, no finishSelect will be called.
+     */
+    default ConnectorTableHandle beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
How would you solve the use case presented here: https://github.com/prestodb/presto/pull/5122#discussion_r65201305 then?
BTW What's wrong with creating a new TableHandle in place of the former one?
I'm passing the former `ConnectorTableHandle` to `ConnectorMetadata` so you know what was passed previously.
If you didn't return any `ConnectorTableHandle` from `beginSelect`, I'm not changing the former `TableHandle`.